tools/wake/main.cpp

```diff
@@ -381,6 +383,31 @@ int main(int argc, char **argv) {
     return 1;
   }
 
+  // If the user asked us to clean the local build, do so.
+  if (clean) {
+    // Find all the file we need to delete and then close the database
+    auto files = db.get_outputs();
+    db.close();
+
+    // Delete all the files
+    for (const auto& file : files) {
+      if (unlink(file.c_str()) == -1) {
+        if (errno == EISDIR) continue;
+        if (errno == ENOENT) continue;
+        std::cerr << "error: unlink(" << file << "): " << strerror(errno) << std::endl;
+        return 1;
+      }
+    }
```

Comment:
I'm wondering if we should (after file deletes) loop through the directories that wake created and delete any that are empty, leaving the non-empty ones as an obvious 'here is stuff here' indicator

Reply:
That does seem more tidy